scrapy/core/downloader/handlers/s3.py

```diff
@@ -69,14 +67,4 @@ def download_request(self, request, spider):
             self._signer.add_auth(awsrequest)
             request = request.replace(
                 url=url, headers=awsrequest.headers.items())
-        else:
```

Comment:
The `elif` above can become an `else` now, since `self._signer` is never `None` if `self.anon` is `True`.